File: src/main/java/com/prestacaoservicos/entity/Agendamento.java
package com.prestacaoservicos.entity;

import com.prestacaoservicos.enums.StatusAgendamentoEnum;
import jakarta.persistence.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import java.util.Objects;

@Entity
@Table(name = "agendamento")
public class Agendamento {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "cd_agendamento")
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "cd_cliente_user", nullable = false)
    private User cliente;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "cd_prestador_user", nullable = false)
    private User prestador;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "cd_servico", nullable = false)
    private Servico servico;

    @Column(name = "dt_agendamento", nullable = false)
    private LocalDateTime dataHora;

    @Column(name = "vl_agendamento", nullable = false)
    private BigDecimal valor;

    @Enumerated(EnumType.STRING)
    @Column(name = "ds_status", nullable = false, length = 20)
    private StatusAgendamentoEnum status;

    @Column(name = "ds_motivo_cancelamento", length = 255)
    private String motivoCancelamento;

    public Agendamento() {}

    public Agendamento(Long id, User cliente, User prestador, Servico servico, LocalDateTime dataHora, BigDecimal valor, StatusAgendamentoEnum status, String motivoCancelamento) {
        this.id = id;
        this.cliente = cliente;
        this.prestador = prestador;
        this.servico = servico;
        this.dataHora = dataHora;
        this.valor = valor;
        this.status = status;
        this.motivoCancelamento = motivoCancelamento;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public User getCliente() {
        return cliente;
    }

    public void setCliente(User cliente) {
        this.cliente = cliente;
    }

    public User getPrestador() {
        return prestador;
    }

    public void setPrestador(User prestador) {
        this.prestador = prestador;
    }

    public Servico getServico() {
        return servico;
    }

    public void setServico(Servico servico) {
        this.servico = servico;
    }

    public LocalDateTime getDataHora() {
        return dataHora;
    }

    public void setDataHora(LocalDateTime dataHora) {
        this.dataHora = dataHora;
    }

    public BigDecimal getValor() {
        return valor;
    }

    public void setValor(BigDecimal valor) {
        this.valor = valor;
    }

    public StatusAgendamentoEnum getStatus() {
        return status;
    }

    public void setStatus(StatusAgendamentoEnum status) {
        this.status = status;
    }

    public String getMotivoCancelamento() {
        return motivoCancelamento;
    }

    public void setMotivoCancelamento(String motivoCancelamento) {
        this.motivoCancelamento = motivoCancelamento;
    }

    @Override
    public boolean equals(Object o) {
        if (o == null || getClass() != o.getClass()) return false;
        Agendamento that = (Agendamento) o;
        return Objects.equals(id, that.id) && Objects.equals(cliente, that.cliente) && Objects.equals(prestador, that.prestador) && Objects.equals(servico, that.servico) && Objects.equals(dataHora, that.dataHora) && Objects.equals(valor, that.valor) && status == that.status && Objects.equals(motivoCancelamento, that.motivoCancelamento);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, cliente, prestador, servico, dataHora, valor, status, motivoCancelamento);
    }

    @Override
    public String toString() {
        return "Agendamento{" +
                "id=" + id +
                ", cliente=" + cliente +
                ", prestador=" + prestador +
                ", servico=" + servico +
                ", dataHora=" + dataHora +
                ", valor=" + valor +
                ", status=" + status +
                ", motivoCancelamento='" + motivoCancelamento + '\'' +
                '}';
    }
}

File: internal/domain/entity/agendamento.go
package entity

import (
	"time"
	"github.com/Luiz-Ferndo/API-Servicos-P1/internal/domain/enum"
)

// Agendamento representa um agendamento de serviço
type Agendamento struct {
	ID                  uint                      `gorm:"primaryKey;column:cd_agendamento" json:"id"`
	ClienteID           uint                      `gorm:"not null;column:cd_cliente_user" json:"clienteId"`
	PrestadorID         uint                      `gorm:"not null;column:cd_prestador_user" json:"prestadorId"`
	ServicoID           uint                      `gorm:"not null;column:cd_servico" json:"servicoId"`
	DataHora            time.Time                 `gorm:"not null;column:dt_agendamento" json:"dataHora"`
	Valor               float64                   `gorm:"not null;column:vl_agendamento;type:decimal(10,2)" json:"valor"`
	Status              enum.StatusAgendamento    `gorm:"not null;column:ds_status;type:varchar(20)" json:"status"`
	MotivoCancelamento  *string                   `gorm:"column:ds_motivo_cancelamento;type:varchar(255)" json:"motivoCancelamento,omitempty"`
	Cliente             *User                     `gorm:"foreignKey:ClienteID;references:ID" json:"cliente,omitempty"`
	Prestador           *User                     `gorm:"foreignKey:PrestadorID;references:ID" json:"prestador,omitempty"`
	Servico             *Servico                  `gorm:"foreignKey:ServicoID;references:ID" json:"servico,omitempty"`
}

// TableName especifica o nome da tabela
func (Agendamento) TableName() string {
	return "agendamento"
}
